libs/auth-api-lib/src/lib/delegations/admin/delegation-admin-custom.service.ts

```diff
@@ -1,99 +1,226 @@
+import { DelegationDelegationType } from '../models/delegation-delegation-type.model'
+import { DelegationsIncomingCustomService } from '../delegations-incoming-custom.service'
+import { DelegationValidity } from '../types/delegationValidity'
 
 @Injectable()
 export class DelegationAdminCustomService {
   constructor(
     @InjectModel(Delegation)
     private delegationModel: typeof Delegation,
+    @InjectModel(DelegationDelegationType)
+    private delegationDelegationTypeModel: typeof DelegationDelegationType,
+    private readonly zendeskService: ZendeskService,
     private delegationResourceService: DelegationResourcesService,
+    private delegationsIncomingCustomService: DelegationsIncomingCustomService,
     private delegationIndexService: DelegationsIndexService,
     private delegationScopeService: DelegationScopeService,
+    private namesService: NamesService,
     private sequelize: Sequelize,
   ) {}
 
+  private getNationalIdsFromZendeskTicket(ticket: Ticket): {
+    fromReferenceId: string
+    toReferenceId: string
+  } {
+    const fromReferenceId = ticket.custom_fields.find(
+      (field) => field.id === ZENDESK_CUSTOM_FIELDS.DelegationFromReferenceId,
+    )
+    const toReferenceId = ticket.custom_fields.find(
+      (field) => field.id === ZENDESK_CUSTOM_FIELDS.DelegationToReferenceId,
+    )
+
+    if (!fromReferenceId || !toReferenceId) {
+      throw new Error('Zendesk ticket is missing required custom fields')
+    }
+
+    return {
+      fromReferenceId: fromReferenceId.value,
+      toReferenceId: toReferenceId.value,
+    }
+  }
+
   async getAllDelegationsByNationalId(
     nationalId: string,
   ): Promise<DelegationAdminCustomDto> {
-    const incomingDelegations = await this.delegationModel.findAll({
-      where: {
-        toNationalId: nationalId,
-      },
-      include: [
-        {
-          model: DelegationScope,
-          required: true,
-          include: [
-            {
-              model: ApiScope,
-              as: 'apiScope',
-              required: true,
-              where: {
-                enabled: true,
-              },
-              include: [
-                {
-                  model: ApiScopeDelegationType,
-                  required: true,
-                  where: {
-                    delegationType: AuthDelegationType.Custom,
-                  },
+    const incomingDelegationPromises = []
+
+    incomingDelegationPromises.push(
+      this.delegationsIncomingCustomService.findAllValidIncoming({
+        nationalId: nationalId,
+        validity: DelegationValidity.ALL,
+      }),
+    )
+
+    incomingDelegationPromises.push(
+      this.delegationsIncomingCustomService.findAllValidGeneralMandate({
+        nationalId: nationalId,
+      }),
+    )
+
+    const [
+      incomingDelegationSets,
+      outgoingDelegations,
+      generalOutgoingDelegations,
+    ] = await Promise.all([
+      await Promise.all(incomingDelegationPromises),
```

Comment:
Just flatten this and consistent variable naming

```suggestion
    const [
      incomingCustomDelegations,
      incomingGeneralDelegations
      outgoingCustomDelegations,
      outgoingGeneralDelegations,
    ] = await Promise.all([
      this.delegationsIncomingCustomService.findAllValidIncoming({
        nationalId: nationalId,
        validity: DelegationValidity.ALL,
      }),
      this.delegationsIncomingCustomService.findAllValidGeneralMandate({
        nationalId: nationalId,
      }),
```